services/checks/checks.go

```diff
@@ -565,46 +570,75 @@ type sttCheckResult struct {
 	result    check.Result
 }
 
-func (s *Service) processResults(ctx context.Context, check check.Check, target target, resID string) ([]sttCheckResult, error) {
+// StarlarkScriptData represents the data we need to pass to the binary to run starlark scripts.
+type StarlarkScriptData struct {
+	CheckName    string                   `json:"name"`
+	Script       string                   `json:"script"`
+	CheckVersion uint32                   `json:"version"`
+	ScriptInput  []map[string]interface{} `json:"input"`
```

Comment:
⚠️ 

> pass input as binary protobuf we receive from pmm-agent via stdin

The reason is that JSON supports only a small subset of types we use. For example, information about all specific integer types is lost (all JSON numbers are float64), time.Time type information is lost, etc. So yeah, we really have to pass the binary data there. That would require you to move `agentpb.UnmarshalActionQueryResult` call out of `waitForResult` to pmm-managed-starlark